src/fal/packages/dependency_analysis.py

```diff
@@ -1,45 +1,26 @@
 from __future__ import annotations
 
-import glob
-import importlib
-import importlib.util
-import os
+import importlib_metadata
 from pathlib import Path
 from typing import Iterator, Optional, Tuple
 
 
-# NOTE: This is from dbt https://github.com/dbt-labs/dbt-core/blob/7bd861a3514f70b64d5a6c642b4204b50d0d3f7e/core/dbt/version.py#L209-L236
-def _get_dbt_plugins_info() -> Iterator[Tuple[str, str]]:
-    for plugin_name in _get_adapter_plugin_names():
-        if plugin_name == "core":
-            continue
-        try:
-            mod = importlib.import_module(f"dbt.adapters.{plugin_name}.__version__")
-        except ImportError:
-            # not an adapter
-            continue
-        yield plugin_name, mod.version  # type: ignore
-
-
-def _get_adapter_plugin_names() -> Iterator[str]:
-    spec = importlib.util.find_spec("dbt.adapters")
-    # If None, then nothing provides an importable 'dbt.adapters', so we will
-    # not be reporting plugin versions today
-    if spec is None or spec.submodule_search_locations is None:
-        return
-
-    for adapters_path in spec.submodule_search_locations:
-        version_glob = os.path.join(adapters_path, "*", "__version__.py")
-        for version_path in glob.glob(version_glob):
-            # the path is like .../dbt/adapters/{plugin_name}/__version__.py
-            # except it could be \\ on windows!
-            plugin_root, _ = os.path.split(version_path)
-            _, plugin_name = os.path.split(plugin_root)
-            yield plugin_name
+def _get_dbt_packages() -> Iterator[Tuple[str, str]]:
+    # package_distributions will return a mapping of top-level package names to a list of distribution names (
+    # the PyPI names instead of the import names). An example distirbution info is the following, which
+    # contains both the main exporter of the top-level name (dbt-core) as well as all the packages that
+    # export anything to that namespace:
+    #   {"dbt": ["dbt-core", "dbt-postgres", "dbt-athena-adapter"]}
+    #
+    # This won't only include dbt.adapters.xxx, but anything that might export anything to the dbt namespace
+    # (e.g. a hypothetical plugin that only exports stuff to dbt.includes.xxx) which in theory would allow us
+    # to replicate the exact environment.
```

Comment:
Really interesting. I think this sounds fine, but I am curious if there will be complains or anything like that. Let's go with this and we will figure it out!

Reply:
We could also limit this to adapters only, but I don't see any reason for it right now. As you said, if in the future if we see this as a problem we can add that filter as well.